src/tokens/ERC1155/ERC1155Supply.sol

```diff
@@ -0,0 +1,122 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.17;
+
+import {ERC1155} from "@0xsequence/erc-1155/contracts/tokens/ERC1155/ERC1155.sol";
+import {ERC1155SupplyErrors} from "./ERC1155SupplyErrors.sol";
+
+/**
+ * An ERC1155 extension that tracks token supply.
+ */
+contract ERC1155Supply is ERC1155, ERC1155SupplyErrors {
+    // Maximum supply globally and per token. 0 indicates unlimited supply
+    uint256 public totalSupplyCap;
+    mapping(uint256 => uint256) public tokenSupplyCap;
+
+    uint256 public totalSupply;
+    mapping(uint256 => uint256) public tokenSupply;
+
+    /**
+     * Mint _amount of tokens of a given id
+     * @param _to The address to mint tokens to
+     * @param _id Token id to mint
+     * @param _amount The amount to be minted
+     * @param _data Data to pass if receiver is contract
+     */
+    function _mint(address _to, uint256 _id, uint256 _amount, bytes memory _data) internal {
+        // Check supply cap
+        totalSupply += _amount;
+        if (totalSupplyCap > 0 && totalSupply > totalSupplyCap) {
```

Comment:
save `totalSupplyCap` as a local variable, this will get called a lot and it doesn't make sense to load from storage twice